src/BBT.MaybePattern.Tests/TestUtils.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace BBT.MaybePattern.Tests
+{
+    using System.IO;
+    using System.Runtime.Serialization.Formatters.Binary;
+
+    /// <summary>
+    /// Provides utility method for test purposes.
+    /// </summary>
+    public class TestUtils
+    {
+        /// <summary>
+        /// Serializes the given object into memory stream.
+        /// </summary>
+        /// <param name="objectType">the object to be serialized.</param>
+        /// <returns>The serialized object as memory stream.</returns>
+        public static MemoryStream SerializeToStream(object objectType)
+        {
+            var stream = new MemoryStream();
```

Comment:
Use try/catch to dispose stream in case an exception is thrown before stream could be returned:

```
var stream = new MemoryStream();
try
{
    var lFormatter = new BinaryFormatter();
    lFormatter.Serialize(stream, objectType);
    return stream
}
catch
{
    if (stream != null)
    {
        stream.Dispose();
    }

    throw;
}
```